src/test/java/uk/gov/justice/hmpps/casenotes/services/CaseNoteServiceTest.kt

```diff
@@ -104,14 +111,56 @@ class CaseNoteServiceTest {
           "authorUserId",
           "eventId",
           "sensitive",
+          "systemGenerated",
         )
         .isEqualTo(nomisCaseNote)
       assertThat(caseNote.text).isEqualTo("original")
       assertThat(caseNote.authorUserId).isEqualTo("23456")
       assertThat(caseNote.locationId).isEqualTo("agency")
       assertThat(caseNote.caseNoteId).isEqualTo("12345")
       assertThat(caseNote.eventId).isEqualTo(12345)
-      Mockito.verify(caseNoteTypeRepository).findCaseNoteTypeByParentTypeTypeAndType("type", "SUB")
+      assertThat(caseNote.systemGenerated).isTrue()
+      Mockito.verify(caseNoteTypeRepository).findCaseNoteTypeByParentTypeTypeAndType("ACP", "POS1")
+    }
+
+    @Test
+    fun `non sensitive and non system generated case note stored in NOMIS`() {
```

Comment:
These two NOMIS tests confirm that a null system generated request value sets system generated based on the sub type. I think one final test that confirms that if the request contains systemGenerated = true and the sub type is dpsUserSelectable, it uses the request value